src/System.Data.Common/ref/System.Data.Common.cs

```diff
@@ -1939,58 +1939,85 @@ public abstract partial class DbDataReader : System.MarshalByRefObject, System.C
         public abstract object this[int ordinal] { get; }
         public abstract object this[string name] { get; }
         public abstract int RecordsAffected { get; }
-        public virtual int VisibleFieldCount { get { throw null; } }
+        public virtual int VisibleFieldCount => throw null;
```

Comment:
As already discussed above, the entire file is full of the older style, without expression-bodied methods/properties. I don't think we should switch to the new style - even for the newly-introduced overloads - to maintain consistency. We can always do a separate PR later to convert the entire file.

Reply:
@roji, what about the new file that holds the new methods; `System.Data.Common.netcoreapp.cs`? Does it need to follow the other file or can it use `=> throw null`?

Reply:
I'd say the new file can be all new syntax because it's a new file it won't hurt readability whien reviewing the changes here. Mixing changes in a PR makes it harder to see which ones relate to specific tasks. You want to try and minimize the change set as much as possible which means usually you end up with small and frequent PR's.

Reply:
Agree with @Wraith2. In any case the idea would be to some cleanup in a later PR and simplify all of System.Data - it's just better to keep it out of this PR.